NEWS.md

```diff
@@ -3,6 +3,306 @@ XGBoost Change Log
+
+### Allow individual workers to recover from faults (#4808, #4966)
+* Status quo: if a worker fails, all workers are shut down and restarted, and learning resumes from the last checkpoint. This involves requesting resources from the scheduler (e.g. Spark) and shuffling all the data again from scratch. Both of these operations can be quite costly and block training for extended periods of time, especially if the training data is big and the number of worker nodes is in the hundreds.
+* The proposed solution is to recover the single node that failed, instead of shutting down all workers. The rest of the clusters wait until the single failed worker is bootstrapped and catches up with the rest.
+* See roadmap at #4753. Note that this is work in progress. In particular, the feature is not yet available from XGBoost4J-Spark.
+
+### Accurate prediction for DART models
+* Use DART tree weights when computing SHAPs (#5050)
+* Don't drop trees during DART prediction by default (#5115)
+* Fix DART prediction in R (#5204)
+
+### Make external memory more robust
+* Fix issues with training with external memory on cpu (#4487)
+* Fix crash with approx tree method on cpu (#4510)
+* Fix external memory race in `exact` (#4980). Note: `dmlc::ThreadedIter` is not actually thread-safe. We would like to re-design it in the long term.
+
+### Major refactoring of the `DMatrix` class (#4686, #4744, #4748, #5044, #5092, #5108, #5188, #5198)
+* Goal 1: improve performance and reduce memory consumption. Right now, if the user trains a model with a NumPy array as training data, the array gets copies 2-3 times before training begins. We'd like to reduce duplication of the data matrix.
+* Goal 2: Expose a common interface to external data, unify the way DMatrix objects are constructed and simplify the process of adding new external data sources. This work is essential for ingesting cuPy arrays.
+* RFC: #4354, Roadmap: #5143
+* This work is also relevant to external memory support on GPUs.
+
+### Breaking: XGBoost Python package now requires Python 3.5 or newer (#5021, #5274)
+* Python 3.4 has reached its end-of-life on March 16, 2019, so we now require Python 3.5 or newer.
+
+### Breaking: GPU algorithm now requires CUDA 9.0 and higher (#4527, #4580)
+
+### Breaking: `n_gpus` parameter removed; multi-GPU training now requires a distributed framework (#4579, #4749, #4773, #4810, #4867, #4908)
+* #4531 proposed removing support for single-process multi-GPU training. Contributors would focus on multi-GPU support through distributed frameworks such as Dask and Spark, where the framework would be expected to assign a worker process for each GPU independently. By delegating GPU management and data movement to the distributed framework, we can greatly simplify the core XGBoost codebase, make multi-GPU training more robust, and reduce burden for future development.
+
+### Breaking: Some deprecated features have been removed
```

Comment:
Sorry I think I forgot to make a breaking tab.  For fixing dart booster, we added a `training` parameter to prediction to distinguish whether the prediction result is used for training.  This breaks C API.  But higher level language bindings are unaffected by using default parameter value.